src/components/views/Booktrak/BooktrakListings.tsx

```diff
@@ -4,98 +4,158 @@ import React, { useState, useEffect } from "react";
 // Redux/ Routing imports
 import { useAppSelector } from "../../../lib/store";
 import { getWSO } from "../../../lib/authSlice";
-import { Link, useNavigate } from "react-router-dom";
+import { useNavigate } from "react-router-dom";
 
 // Additional Imports
 import {
   ModelsBook,
   ModelsBookListing,
 } from "wso-api-client/lib/services/types";
-import { BookConditionEnumToString } from "./BooktrakUtils";
+import PaginationButtons from "../../PaginationButtons";
+import BooktrakListingsTable from "./BooktrakListingsTable";
+import "../../stylesheets/Booktrak.css";
+import { ModelsBookListingWithTitle } from "./BooktrakUtils";
 
-const BooktrakListings = ({ book }: { book: ModelsBook }) => {
+type ServerError = {
+  errorCode: number;
+  message: string;
+};
+
+function isServerError(error: unknown): error is ServerError {
+  return (
+    typeof error === "object" &&
+    error !== null &&
+    "errorCode" in error &&
+    "message" in error
+  );
+}
+
+const ListingTypeEnum = ModelsBookListing.ListingTypeEnum;
+const BooktrakListings = ({
+  book,
+  onlyShowBuyListings,
+  displayBuyAndSell,
+}: {
+  book?: ModelsBook;
+  onlyShowBuyListings?: boolean;
+  displayBuyAndSell?: boolean;
+}) => {
   const wso = useAppSelector(getWSO);
   const navigateTo = useNavigate();
-  const [buyListings, updateBuyListings] = useState<ModelsBookListing[]>([]);
-  const [sellListings, updateSellListings] = useState<ModelsBookListing[]>([]);
+  const [listings, updateListings] = useState<ModelsBookListingWithTitle[]>([]);
+  const [buyListings, updateBuyListings] = useState<
+    ModelsBookListingWithTitle[]
+  >([]);
+  const [sellListings, updateSellListings] = useState<
+    ModelsBookListingWithTitle[]
+  >([]);
+
+  const [currentPage, updateCurrentPage] = useState(0);
+  const maxListingsPerPage = 20;
```

Comment:
Use the backend pagination to paginate. You can use the `offset` field in `listBookListings`